backend/tests/unit/services/test_metrics_collector.py

```diff
@@ -0,0 +1,186 @@
+# Copyright (C) 2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import time
+from datetime import UTC, datetime, timedelta
+from multiprocessing.shared_memory import SharedMemory
+from unittest.mock import patch
+from uuid import uuid4
+
+from app.services.metrics_collector import MAX_MEASUREMENTS, SHM_NAME, SIZE, MetricsCollector
+
+SHM = SharedMemory(name=SHM_NAME, create=True, size=SIZE)
```

Comment:
This should be proprely cleaned up explicitly.
We can use `scope=module` fixture in autouse mode, so that cleanup can be handled after `yield`, e.g.:

```python
@pytest.fixture(scope="module", autouse=True)
def fxt_metrics_mem():
  shm = SharedMemory(name=SHM_NAME, create=True, size=SIZE)
  yield
  shm.close()
  shm.unlink()
```